.github/workflows/release.yml

```diff
@@ -13,16 +13,45 @@ on:
         required: false
 
 jobs:
+  prepare_framework:
+    runs-on: macos-13-xlarge
+    name: 'Create XCFramework'
+    steps:
+      - uses: actions/checkout@v4
+    
+      - name: 'Generate XCFramework'
+        run: |
+          ./scripts/ci-select-xcode.sh 15.2
+          make build-xcframework
```

Comment:
`m`: I think it would be good to explain why we need to build the XCFramework here with a comment, and can't rely on building it in `build.yml.

Reply:
```suggestion
      - name: 'Generate XCFramework'
        run: |
          ./scripts/ci-select-xcode.sh 15.2
          #We need to build the framework during release to get it's SHA value
          #the framework will be saved as an artefact and we will use the same
          #binary for the entire release process to avoid the SHA to change
          make build-xcframework 
```